Tests/scripts/secrets.py

```diff
@@ -0,0 +1,223 @@
+    :param files_string: string representing the git diff files
+    :return: text_files_list: string of full path to text files
+    """
+    # file statuses to filter from the diff, no need to test deleted files.
+    accepted_file_statuses = ['M', 'A']
+    all_files = files_string.split('\n')
+    text_files_list = set()
+    for file_name in all_files:
+        file_data = file_name.split()
+        if not file_data:
+            continue
+
+        file_status = file_data[0]
+        file_path = file_data[1]
+        # only modified/added file, text readable, exclude white_list file
+        if file_status.upper() in accepted_file_statuses and is_text_file(file_path) \
+                and SECRETS_WHITE_LIST_FILE not in file_path:
+            text_files_list.add(file_path)
+
+    return text_files_list
+
+
+def get_all_diff_text_files(branch_name, is_circle):
+    """
+    :param branch_name: current branch being worked on
+    :param is_circle: boolean to check if being ran from circle
+    :return:
+    """
+    if is_circle:
+        branch_changed_files_string = \
+            validate_files_structure.run_git_command("git diff --name-status origin/master...{}".format(branch_name))
+        text_files_list = get_diff_text_files(branch_changed_files_string)
+
+    else:
+        local_changed_files_string = validate_files_structure.run_git_command("git diff --name-status --no-merges HEAD")
+        text_files_list = get_diff_text_files(local_changed_files_string)
+
+    return text_files_list
+
+
+def search_potential_secrets(secrets_file_paths):
+    """Returns potential secrets(sensitive data) found in committed and added files
+    :param secrets_file_paths: paths of files that are being commited to git repo
+    :return: dictionary(filename: (list)secrets) of strings sorted by file name for secrets found in files
+    """
+    # Get generic white list set
+    with io.open('./Tests/secrets_white_list.json', mode="r", encoding="utf-8") as secrets_white_list_file:
+        secrets_white_list = set(json.load(secrets_white_list_file))
+
+    secrets_found = {}
+
+    for file_path in secrets_file_paths:
+        file_name = file_path.split('/')[-1]
+        high_entropy_strings = []
+        regex_secrets = []
+        yml_file_contents = None
+        skip_secrets = False
+
+        # if py file, search for yml in order to retrieve temp white list
+        file_path_temp, file_extension = os.path.splitext(file_path)
+        if file_extension == '.py':
+            yml_file_contents = retrieve_related_yml(file_path_temp)
+
+        # Open each file, read its contents in UTF-8 encoding to avoid unicode characters
+        with io.open('./' + file_path, mode="r", encoding="utf-8") as file:
+            file_contents = file.read()
+
+            # Add all context output paths keywords to whitelist temporary
+            temp_white_list = create_temp_white_list(yml_file_contents if yml_file_contents else file_contents)
```

Comment:
we either parse the yml or just the yml part from a package, there is not both since if you have yml_file_contents the main file you are parsing is not yml if you it is yml you won't have yml_file_contents